src/components/TeamMemberTasks/components/TaskCompletedModal.jsx

```diff
@@ -35,41 +39,46 @@ const TaskCompletedModal = React.memo(props => {
     });
 
     const updatedTask = { ...task, resources: newResources };
-    props.updateTask(task._id, updatedTask);
+    props.updateTask(task._id, updatedTask, setIsLoadingTask);
   };
 
   useEffect(() => {
     loadUserTasks(props.userId);
-  }, [props.userID, props.tasks]);
+  }, [props.userId]);
 
   return (
     <Modal isOpen={props.isOpen} toggle={() => props.popupClose()}>
       <ModalHeader toggle={() => props.popupClose()}>Mark as Done</ModalHeader>
-      <ModalBody>
-        <p>Are you sure you want to mark this task as done?</p>
-        <ModalFooter>
-          <Button
-            color="primary"
-            onClick={() => {
-              removeTaskFromUser(props.task);
-              props.setClickedToShowModal(false);
-              props.setCurrentUserId('');
-              closePopup();
-            }}
-          >
-            Confirm
-          </Button>
-          <Button
-            onClick={() => {
-              props.setClickedToShowModal(false);
-              props.setCurrentUserId('');
-              closePopup();
-            }}
-          >
-            Cancel
-          </Button>
-        </ModalFooter>
-      </ModalBody>
+      {isLoadingTask ? (
+        <ModalBody>
+          <p>Loading...</p>
+        </ModalBody>
+      ) : (
+        <ModalBody>
+          <p>Are you sure you want to mark this task as done?</p>
+          <ModalFooter>
+            <Button
+              color="primary"
+              onClick={() => {
+                setIsLoadingTask(true);
+                removeTaskFromUser(props.task);
```

Comment:
Does this also mark the task as complete on the tasks page?

Reply:
It needs further investigation to answer this.